### PR TITLE
Adds documentation for the new ACL APIs

### LINKED ISSUE #4834: Document new ACL HTTP API
All of the new endpoints need to be documented.

### PR DESCRIPTION
**Pull request by mkeeler**
fixes #4834 

Also sneaking in a couple minor fixes:

* Ensures the anonymous token is created with the CreateTime set
* The rule translation endpoint now requires `acl:read` permissions to use.

### CODE REVIEW

**Comment by pearkes** [comment]
TODO: Should add a section to the main ACL guide regarding how it behaves with multi-dc (replication).

File: agent/acl_endpoint.go [hunk 1]
```diff
@@ -102,6 +102,18 @@ func (s *HTTPServer) ACLRulesTranslate(resp http.ResponseWriter, req *http.Reque
 		return nil, nil
 	}
 
+	var token string
+	s.parseToken(req, &token)
+	rule, err := s.agent.resolveToken(token)
+	if err != nil {
+		return nil, err
+	}
+	// Should this require lesser permissions? Really the only reason to require authorization at all is
```
[the remark this answers is not in the document]
Comment (banks) [reply agent/acl_endpoint.go]: Lets leave it like this. The DOS vector doesn't sound super likely to me but it's easier to be more restrictive now than later and anyone using UI or CLI to manage ACLs already needs this anyway.

File: website/source/layouts/api.erb [hunk 2]
```diff
@@ -10,8 +10,21 @@
 
       <hr>
 
-      <li<%= sidebar_current("api-acls") %>>
-        <a href="/api/acl.html">ACLs</a>
+      <li<%= sidebar_current("api-acl") %>>
```
[the remark this answers is not in the document]
Comment (kaitlincarter-hc) [reply website/source/layouts/api.erb]: Agreed. I'd also like to call Legacy "Legacy Tokens" to be extra clear.

File: agent/acl_endpoint.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pearkes) [reply agent/acl_endpoint.go]: 👍

File: website/source/api/acl/acl.html.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+# ACL HTTP API
+
+The `/acl` endpoints are used to manage ACL tokens and policies in Consul, [bootstrap the ACL system](#bootstrap-acls), [check ACL replication status](#check-acl-replication), and [translate rules](#translate-rules). There are additional pages for managing [tokens](/api/acl/tokens.html) and [policies](/api/acl/policies.html) with the `/acl` endpoints.
+
+For more information about ACLs, please see the [ACL Guide](/docs/guides/acl.html).
+
+## Bootstrap ACLs
+
+This endpoint does a special one-time bootstrap of the ACL system, making the first
+management token if the [`acl.tokens.master`](/docs/agent/options.html#acl_tokens_master)
```
[the remark this answers is not in the document]
Comment (kaitlincarter-hc) [reply website/source/api/acl/acl.html.md]: Maybe we can revisit in the future.

[the remark this answers is not in the document]
Comment (pearkes) [reply website/source/api/acl/acl.html.md]: Agree.

**Pull request merged**